velox/type/DecimalUtil.h

```diff
@@ -147,6 +147,36 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleFloating(
+      const TInput inputValue,
+      const int toPrecision,
+      const int toScale) {
+    if (!std::isfinite(inputValue)) {
+      return std::nullopt;
+    }
+    auto toValue =
```

Comment:
Here may overflow, please test the double::max

Reply:
Yes, it will overflow here, so I have checked below.

Reply:
if the inputValue is double::max, you check the unexpected overflowd double value, not the value after multiply.

Reply:
If the value after multiplication is overflowed, the 'toValue' is std::inf or something like this. It will not pass the check below.
But indeed we could set the default value of 'overflow' by checking if 'toValue' is overflowed.

done.